hw/ip/spi_host/data/spi_host.hjson

```diff
@@ -0,0 +1,590 @@
+               The most significant byte (MSB) of TXDATA is transmitted first, and
+               received data is placed in the MSB location of RXDATA.  If ByteOrder = 1,
+               a Little-Endian ordering is used for these registers, and the LSB of each
+               gets priority for receiving and transmitting data.'''
+      type: "int",
+      default: "0"
+    },
+    { name: "MaxCS",
+      desc: "The number of active-low chip select (cs_n) lines to create. (Maximum 8)",
+      type: "int",
+      default: "8"
+    },
+  ],
+  available_output_list: [
+    { name:  "sck_o"
+      desc:  "SPI Clock"
+    },
+    { name:  "cs_no"
+      desc:  '''Chip Enable# (One hot, active low).  The size of this port should match MaxCS'''
+      width: "8"
+    },
+    { name:  "sd_o",
+      desc:  "SPI data IO outputs",
+      width: "4"
+    },
+    { name:  "sd_t",
+      desc:  "SPI data tri-state control",
+      width: "4"
+    }
+  ],
+  available_input_list: [
+    { name:  "sd_i",
+      desc:  "SPI data IO inputs",
+      width: "4"
+    },
+  ],
+  interrupt_list: [
+    { name: "error",
+      desc: '''Error-related interrupts, see ERROR_MASK register for more
+               information.'''
+    },
+    { name: "event",
+      desc: '''Event-related interrupts, see EVENT_MASK register for more
+               information.'''
+    }
+  ],
+  inter_signal_list: [
+    { struct:  "logic",
+      type:    "unit",
+      name:    "pt_sck_i",
+      act:     "rcv",
+      width:   1,
+      package: ""
+    },
+    { struct:  "logic",
+      type:    "unit",
+      name:    "pt_cs_ni",
+      act:     "rcv",
+      width:   8,
+      package: ""
+    },
+    { struct:  "logic",
+      type:    "unit",
+      name:    "pt_sdo_i",
+      act:     "rcv",
+      width:   4,
+      package: ""
+    },
+    { struct:  "logic",
+      type:    "unit",
+      name:    "pt_sdt_i",
+      act:     "rcv",
+      width:   4,
+      package: ""
+    },
+    { struct:  "logic",
+      type:    "unit",
+      name:    "pt_sdi_o",
+      act:     "req",
+      width:   4,
+      package: ""
+    },
+  ]
+  registers: [
+    { name: "CONTROL",
+      desc: "Control register",
+      swaccess: "wo",
+      hwaccess: "hro",
+      fields: [
+        { bits: "31",
+          name: "SPIEN",
+          desc: '''Enables the SPI host.  On reset, this field is 0, meaning that no transactions can proceed.'''
+          resval: "0x0"
+        },
+        { bits: "30",
+          name: "RST_FSM",
+          desc: '''This bit holds the control FSM in reset when set to 1.'''
+          resval: "0x0",
+          hwqe: "true",
+          hwacces: "hrw"
+        },
+        { bits: "29",
+          name: "RST_TXFIFO",
+          desc: '''Resets the TX FIFO when set to 1.'''
+          resval: "0x0",
+        },
+        { bits: "28",
+          name: "RST_RXFIFO",
+          desc: '''Resets pulse the RX FIFO when set to 1.'''
```

Comment:
Stray "pulse" here? Also, I read "set" here as "the corresponding bit of the register is set". In that case, it doesn't quite seem like the right word, because a write of 1 is transient: it doesn't set a bit anywhere. Maybe "when written as 1"?

Reply:
Ah this was originally meant to be a single pulse, but for now software has to undo it manually.  Stray word removed.